### PR TITLE
docs(a11y): add section on testing for a11y

### LINKED ISSUE #528: Update a11y contributor guidelines
The contributor guidelines should be updated to reflect the latest recommendations for tools. 

As part of the [tooling added to check for a11y violations](https://github.com/patternfly/patternfly-next/pull/517), the contributor guidelines should include some instruction on how to set that up, as well as expectations on what to do when violations are found. During this initial exploration phase, some of the ways we handle violations could be one of the following:

- remove the rule if it isn't relevant for the type of example (e.g. components aren't going to have landmark roles, but page layouts would)
- fix the violation
- open an issue to further investigate the violation if it isn't obvious how to fix or move forward with it.

### PR DESCRIPTION
**Pull request by jgiardino**
Closes #528  

Adds section to README.md on how to test for accessibility. Moves the section added in #615 down to this section and explains how to handle the violations that are flagged. 

Also adds a bullet to the Accessibility Guide in the Testing section to recommend aXe and link back to the README.md.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,6 +53,33 @@ To run the a11y audit locally
...
+PatternFly uses [aXe: The Accessibility Engine](https://www.deque.com/axe/) to check for accessibility violations. Our goal is to meet WCAG 2.0 AA requirements, as noted in our [Accessibility Guide](https://pf-next.com/accessibility-guide).
+
+### How to Perform an Accessibility Audit with aXe
+aXe is available as either a browser extension or npm script.
+
+To run the a11y audit locally:
+- install the latest [chromedriver](http://chromedriver.chromium.org/downloads) and ensure its available on your system `$PATH`
+  - alternatively, macOS users can simply `brew cask install chromedriver`
+- run `npm run dev`
+- run `npm run a11y` (in a separate terminal)
```
Comment (seanforyou23) [inline README.md]: "in a separate terminal" may be better written as "in another console"

File: README.md [hunk 2]
```diff
@@ -57,6 +53,33 @@ To run the a11y audit locally
...
+
+The tool is configured to return WCAG 2.0 AA violations for the full page renderings of all components, layouts, utilities, and demos. The tool will provide feedback about what the violation is and a link to documentation about how to address the violation.
+
+The same tool is also available as a browser extension for [Chrome](https://chrome.google.com/webstore/detail/axe/lhdoppojpmngadmnindnejefpokejbdd) and [Firefox](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/).
```
Comment (seanforyou23) [inline README.md]: Should we also add a link here for axe-coconut? Helps if users want to test against leading edge of axe-core which has experimental support for latest and greatest a11y rules, including support for things like shadow DOM

https://chrome.google.com/webstore/detail/axe-coconut/iobddmbdndbbbfjopjdgadphaoihpojp

File: src/site/pages/accessibility-guide.md [hunk 3]
```diff
@@ -141,14 +141,15 @@ The [WCAG 2.0 techniques](https://www.w3.org/TR/WCAG20-TECHS/Overview.html#conte
...
+Many accessibility issues can be found by doing a few simple checks:
+1. Use an accessibility audit tool to check for violations. If you are using PatternFly in your project, we recommend using [aXe: The Accessibility Engine](https://www.deque.com/axe/) to check for accessibility violations. If you are contributing to PatternFly, refer to our [README.md](https://github.com/patternfly/patternfly-next/blob/master/README.md) on how to run this tool.
+2. Test keyboard accessibility, and check that these requirements are met:
     - All functionality is keyboard accessible
     - Order of elements in the HTML and in the layout follow a logical order
     - Elements with focus are clearly visible
-2. Use the [WAVE browser extension from WebAIM](https://wave.webaim.org/extension/) to disable styles, then test the information architecture and presence of adequate text labels.
-3. Test with any screen reader available in your operating system
-4. Check color contrast for:
+3. Use the [WAVE browser extension from WebAIM](https://wave.webaim.org/extension/) to disable styles, then test the information architecture and presence of adequate text labels.
```
Comment (seanforyou23) [inline src/site/pages/accessibility-guide.md]: There are many ways to disable page styles using built in browser/os settings, that said, it's slightly different depending on which browser and OS a user is running. Do you think it would be better to recommend a method of disabling styles using built in browser/system features, or suggest installing yet another browser plugin?

For example, see the comment in this SO article that illustrates a few ways of doing the same thing. https://stackoverflow.com/a/15280065/4118955

Just thinking it may be a bit much to recommend users install two separate extensions, but maybe it's not that big a deal? Do we need to instruct users how to do this in the first place? Some users may already be using other tools to do this same purpose like standard "Web Developer Tools" in chrome or other similar extensions. The less we have to explain the easier it becomes to quickly scan this document so just trying to trim any language where possible.

File: README.md (hunk 2 above)
Comment (jgiardino) [reply README.md]: I'm on the fence on this. In the README, I'm inclined to only reference tools that we expect to be used as part of the contribution process. Can we chat outside this PR about the benefits of referencing this in the a11y guide?

File: src/site/pages/accessibility-guide.md (hunk 3 above)
Comment (jgiardino) [reply src/site/pages/accessibility-guide.md]: This is a good point. I'll take a stab at this. I'm definitely trying to keep guidelines trim and easy to follow. I do see a difference between the README vs the A11y Guide. The README includes what we expect when you are contributing. The A11y Guide is meant to be informational, regarding what our goals are for patternfly accessibility, and with information to help guide both contributors and consumers when working on accessibility.

**Comment by jgiardino** [comment]
@seanforyou23 @matthewcarleton 

Thanks for your review! I made updates based on @seanforyou23's feedback. I also listed the screen readers that I mentioned on slack this week, with links to keyboard shortcuts for these screen readers.

Let me know if you have any further suggestions.

Thanks!

File: README.md [hunk 4]
```diff
@@ -57,6 +53,33 @@ To run the a11y audit locally
...
+
+The same tool is also available as a browser extension for [Chrome](https://chrome.google.com/webstore/detail/axe/lhdoppojpmngadmnindnejefpokejbdd) and [Firefox](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/). 
+
+### Fixing Violations
+
+Ignore the violations that aren’t related to your contribution.
+
+Fix violations related to your contribution.
+
+If there are violations that are not obvious how to fix, then create an issue with information about the violation that was found. For example, some violations might require further discussion before they can be addressed. And some violations may not be valid and require changes to the workspace or tooling to stop flagging the violation.
```
Comment (michael-coker) [inline README.md]: Should we link "[create an issue](https://github.com/patternfly/patternfly-next/issues/new)"?

File: src/site/pages/accessibility-guide.md [hunk 5]
```diff
@@ -141,14 +141,18 @@ The [WCAG 2.0 techniques](https://www.w3.org/TR/WCAG20-TECHS/Overview.html#conte
 
 
 ## Testing
-To keep testing simple and easy to complete, we ask that contributors complete the following tests to try to catch most of the accessibility issues that may be present:
-1. Test keyboard accessibility, and check that these requirements are met:
+Many accessibility issues can be found by doing a few simple checks:
+1. Use an accessibility audit tool to check for violations. If you are using PatternFly in your project, we recommend using [aXe: The Accessibility Engine](https://www.deque.com/axe/) to check for accessibility violations. If you are contributing to PatternFly, refer to our [README.md](https://github.com/patternfly/patternfly-next/blob/master/README.md) on how to run this tool.
```
Comment (michael-coker) [inline src/site/pages/accessibility-guide.md]: should the `README.md` link go to the "testing for a11y" anchor? https://github.com/jgiardino/patternfly-next/tree/a11y-testing#testing-for-accessibility

**Comment by jgiardino** [comment]
Thanks for the great suggestions, @michael-coker! I made those updates. I think this is ready to merge.
